hedera-node/src/main/java/com/hedera/services/fees/calculation/token/txns/TokenBurnResourceUsage.java

```diff
@@ -47,10 +47,15 @@ public boolean applicableTo(TransactionBody txn) {
 
 	@Override
 	public FeeData usageGiven(TransactionBody txn, SigValueObj svo, StateView view) throws InvalidTxBodyException {
-		Optional<TokenType> tokenType = view.tokenType(txn.getTokenBurn().getToken());
+		final var target = txn.getTokenBurn().getToken();
+		Optional<TokenType> tokenType = view.tokenType(target);
 		SubType subType = subtypeHelper.determineTokenType(tokenType);
 		var sigUsage = new SigUsage(svo.getTotalSigCount(), svo.getSignatureSize(), svo.getPayerAcctSigCount());
 		var estimate = factory.apply(txn, sigUsage).givenSubType(subType);
+		if (subType == SubType.TOKEN_NON_FUNGIBLE_UNIQUE) {
+			final var serialNumsCount = txn.getTokenBurn().getSerialNumbersCount();
+			estimate.givenSerialNumsCount(serialNumsCount);
```

Comment:
The count here should be the remaining count of serial numbers (the count from the `StateView` minus the count of NFTs to be burned)

Reply:
Hmm I don't follow that...the usage will depend **only** on how many serial number appear in the gRPC `TokenBurnTransactionBody`.

Reply:
@tinker-michaelj, @Neeharika-Sompalli, in that case, we should remove the function `givenSerialNumsCount()` in `hapi-fees/src/main/java/com/hedera/services/usage/token/TokenBurnUsage.java` and let the count retrieved from the op. The reason we need to expose `givenSubType()` to set the correct SubType is because the op doesn't carry this info and we need to retrieve it from the current `StateView`.